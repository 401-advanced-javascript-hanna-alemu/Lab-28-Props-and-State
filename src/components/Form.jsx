import React from "react";

class Form extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      name: this.props.name,
      count: this.props.count
    };
  }

  handleChange = event => {
    //Hanna - this function changes the UI state when something is typed
    //event.preventDefault();

    this.setState({
      name: event.target.value
    });
  };

  //Hanna - this function updates the display when the button is pressed, adds one to counter
  handleSubmit = event => {
    event.preventDefault();

    this.setState({
      name: this.state.name,
      count: this.state.count + 1
    });
  };

  render() {
    return (
      <React.Fragment>
        <form onSubmit={this.handleSubmit}>
          <input
            type="text"
            value={this.state.name}
            onChange={this.handleChange}
          />
          <button type="submit">Change {this.props.name} name</button>
        </form>

        <div>
          <div>Name:{this.state.name} </div>
          <div># of Updates: {this.state.count} </div>
        </div>
      </React.Fragment>
    );
  }
}

export default Form;
