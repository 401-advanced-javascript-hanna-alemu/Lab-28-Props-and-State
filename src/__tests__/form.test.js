import React from 'react';

import renderer from 'react-test-renderer';

import Enzyme, { shallow, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

import Form from '../components/Form';

Enzyme.configure({ adapter: new Adapter() });
 

describe('< Form />', () => {
  it('Is rendered when app starts', () => {
    const app = shallow(<Form />);

    expect(app.find('input').exists()).toEqual(true);
    expect(app.find('button').exists()).toEqual(true);

    expect(app.find('input').text()).toEqual('');
  });

  it('name should change when key is pressed', () => {
    const app = mount(< Form />);
    const input = app.find('input');

    input.simulate('key-pressed');

    expect(app.state().name).toEqual(!'');
    
  });
    
  it('count should be toggled when the form is submiitted ', () => {
    const app = mount(< Form />);
    const button = app.find('button');

    button.simulate('submit');

    expect(app.state().count).toEqual(!0);
   
  });
    
  it('matches snapshot', () => {
    const snapshot = renderer.create(< Form />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  
});
