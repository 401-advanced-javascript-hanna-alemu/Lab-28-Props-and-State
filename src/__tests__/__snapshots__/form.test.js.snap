// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< Form /> matches snapshot 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
    />
    <button
      type="submit"
    >
      Change 
       name
    </button>
  </form>,
  <div>
    <div>
      Name:
       
    </div>
    <div>
      # of Updates: 
       
    </div>
  </div>,
]
`;
